File: web/src/model/product.js
/* @flow */
import type { ProductFragment } from '../api/schema'

export type Product = {|
  id: number,
  typeId: number,
  name: string,
  sku: ?string,
  quantity: number,
  sort: number,
  discontinued: boolean,
|}

export function parse({ id, typeId, name, sku, quantity, sort, discontinued }: ProductFragment): Product {
  return {
    id,
    typeId,
    name,
    sku,
    quantity,
    sort,
    discontinued,
  }
}

File: android/app/src/main/kotlin/com/cameldridge/conartist/model/Product.kt
package com.cameldridge.conartist.model

import android.os.Parcelable
import com.cameldridge.conartist.services.api.graphql.fragment.ProductFragment
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Product(
  val id: Id,
  val typeId: Id,
  val name: String,
  val quantity: Int,
  val sort: Int,
  val discontinued: Boolean
): Parcelable {
  companion object {
    fun fromFragment(fragment: ProductFragment) = Product(
      Id.Id(fragment.id),
      Id.Id(fragment.typeId),
      fragment.name,
      fragment.quantity,
      fragment.sort,
      fragment.isDiscontinued
    )
  }
}
